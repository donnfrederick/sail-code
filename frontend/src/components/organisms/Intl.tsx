import React, {
  PropsWithChildren,
  Dispatch,
  SetStateAction,
  useContext,
  useState
} from 'react'
import { IntlProvider } from 'react-intl'
import { messages, browserLocale, Locale } from 'utils/intl'

export type LocaleState = {
  locale: Locale
}

const initialState: LocaleState = {
  locale: browserLocale
}

const IntlStateContext = React.createContext<LocaleState>(initialState)

const SetLocaleStateContext = React.createContext<
  Dispatch<SetStateAction<LocaleState>>
>(() => {
  // empty block
})

export function useLocaleState() {
  return useContext(IntlStateContext)
}
export function useSetLocaleState() {
  return useContext(SetLocaleStateContext)
}

export function LocaleStateProvider(props: {
  initialState?: LocaleState
  children: React.ReactNode
}) {
  const [state, setState] = useState<LocaleState>(
    props.initialState || initialState
  )
  return (
    <IntlStateContext.Provider value={state}>
      <SetLocaleStateContext.Provider value={setState}>
        {props.children}
      </SetLocaleStateContext.Provider>
    </IntlStateContext.Provider>
  )
}

export const Intl = (props: PropsWithChildren<{}>) => {
  const localeState = useLocaleState()

  return (
    <LocaleStateProvider>
      <IntlProvider
        messages={messages[localeState.locale]}
        locale={localeState.locale}
        defaultLocale="en"
      >
        {props.children}
      </IntlProvider>
    </LocaleStateProvider>
  )
}
