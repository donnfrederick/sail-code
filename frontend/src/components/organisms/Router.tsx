import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
