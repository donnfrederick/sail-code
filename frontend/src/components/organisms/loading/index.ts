import Loading from 'components/organisms/loading/Loading'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  isFetchingConversation: boolean
  isFetchingLocations: boolean
  isFetchingNotification: boolean
  isFetchingOrganizations: boolean
  isFetchingStudentsMe: boolean
  isFetchingTeachersMe: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    isFetchingConversation: state.rootReducer.conversation.fetching > 0,
    isFetchingLocations: state.rootReducer.locations.isFetching,
    isFetchingNotification: state.rootReducer.notification.isFetching,
    isFetchingOrganizations: state.rootReducer.organizations.isFetching,
    isFetchingStudentsMe: state.rootReducer.students.isFetching,
    isFetchingTeachersMe: state.rootReducer.teachers.isFetching
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {}
)

export default enhancer(Loading)
