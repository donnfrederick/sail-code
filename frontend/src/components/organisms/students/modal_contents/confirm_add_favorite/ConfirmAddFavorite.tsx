import Button from 'components/atoms/students/Button'
import ModalTextContainer from 'components/molecules/students/ModalTextContainer'
import CompleteAddFavoriteModal from 'components/organisms/students/modal_contents/complete_add_favorite'
import FailFetchingModal from 'components/organisms/students/modal_contents/fail_fetching'
import * as FavoritesModels from 'models/favorites'
import * as React from 'react'
import styled from 'styled-components'

interface Props {
  addFavoritedUserId: number
  authToken: string
  close(): void
  postFavorites(
    authToken: string,
    parameters: FavoritesModels.PostFavoritesRequest
  ): Promise<void>
  setContents(contents: JSX.Element): void
}

export default (props: Props) => {
  const {
    addFavoritedUserId,
    authToken,
    close,
    postFavorites,
    setContents
  } = props

  return (
    <Container>
      <ModalTextContainer heading={'Add this user?'} />
      <Text>
        {"This user's information will be added to your favorite users' list."}
      </Text>
      <ButtonContainer>
        <Button
          type="red"
          text="Cancel"
          width={272}
          height={112}
          fontSize={40}
          onClick={() => close()}
        />
        <Button
          type="white"
          text="OK"
          width={272}
          height={112}
          fontSize={40}
          onClick={async () => {
            if (addFavoritedUserId) {
              const parameters: FavoritesModels.PostFavoritesRequest = {
                user_id: addFavoritedUserId
              }

              try {
                await postFavorites(authToken, parameters)
                setContents(<CompleteAddFavoriteModal />)
              } catch (error) {
                // tslint:disable-next-line
                console.error(error)
                setContents(<FailFetchingModal />)
              }
            }
          }}
        />
      </ButtonContainer>
    </Container>
  )
}

const Container = styled.div`
  width: 100%;
  height: 100%;
  text-align: center;
`

const ButtonContainer = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
`

const Text = styled.div`
  width: 530px;
  margin: 0 auto 50px;
  padding-left: 1em;
  font-size: 24px;
  line-height: 1.38;
  color: #405766;
  text-align: left;
  text-indent: -1em;
`
