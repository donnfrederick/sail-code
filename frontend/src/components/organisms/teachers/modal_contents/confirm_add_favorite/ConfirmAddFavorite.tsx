import Button from 'components/atoms/teachers/Button'
import ModalTextContainer from 'components/molecules/teachers/ModalTextContainer'
import CompleteAddFavoriteModal from 'components/organisms/teachers/modal_contents/complete_add_favorite'
import FailFetchingModal from 'components/organisms/teachers/modal_contents/fail_fetching'
import * as FavoritesModels from 'models/favorites'
import * as React from 'react'
import styled from 'styled-components'

interface Props {
  authToken: string
  addFavoritedUserId: number
  close(): void
  postFavorites(
    authToken: string,
    parameters: FavoritesModels.PostFavoritesRequest
  ): Promise<void>
  setContents(contents: JSX.Element): void
}

export default (props: Props) => {
  const {
    authToken,
    addFavoritedUserId,
    close,
    postFavorites,
    setContents
  } = props

  return (
    <Container>
      <ModalTextContainer heading={'この学生をお気に入りに追加しますか？'} />
      <Text>
        {
          'お気に入りに追加した学生は、[メニュー] > [お気に入り一覧] から確認することができます。'
        }
      </Text>
      <ButtonContainer>
        <Button
          type="white"
          text="いいえ"
          width={272}
          height={112}
          fontSize={40}
          onClick={() => close()}
        />
        <Button
          type="blue"
          text="はい"
          width={272}
          height={112}
          fontSize={40}
          onClick={async () => {
            if (addFavoritedUserId) {
              const parameters: FavoritesModels.PostFavoritesRequest = {
                user_id: addFavoritedUserId
              }

              try {
                await postFavorites(authToken, parameters)
                setContents(<CompleteAddFavoriteModal />)
              } catch (error) {
                // tslint:disable-next-line
                console.error(error)
                setContents(<FailFetchingModal />)
              }
            }
          }}
        />
      </ButtonContainer>
    </Container>
  )
}

const Container = styled.div`
  width: 100%;
  height: 100%;
  text-align: center;
`

const ButtonContainer = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
`

const Text = styled.div`
  width: 530px;
  margin: 0 auto 50px;
  padding-left: 1em;
  font-size: 24px;
  line-height: 1.38;
  color: #405766;
  text-align: left;
  text-indent: -1em;
`
