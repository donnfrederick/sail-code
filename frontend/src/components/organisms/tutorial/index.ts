import Tutorial from 'components/organisms/tutorial/Tutorial'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'
import * as TutorialActions from 'reducers/tutorial'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  shouldShow: boolean
  showConfirmation: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    shouldShow: state.rootReducer.tutorial.shouldShow,
    showConfirmation: state.rootReducer.teachers.showConfirmation
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {
    hide: TutorialActions.hide,
    show: TutorialActions.show
  }
)

export default enhancer(Tutorial)
