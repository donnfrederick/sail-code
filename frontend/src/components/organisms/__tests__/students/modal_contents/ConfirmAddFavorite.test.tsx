import Router from 'components/organisms/Router'
import ConfirmAddFavorite from 'components/organisms/students/modal_contents/confirm_add_favorite'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Intl } from 'components/organisms/Intl'

test('render <ConfirmBlock />', () => {
  const tree = renderer
    .create(
      <Intl>
        <Router>
          <ConfirmAddFavorite addFavoriteUserId={1} />
        </Router>
      </Intl>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
