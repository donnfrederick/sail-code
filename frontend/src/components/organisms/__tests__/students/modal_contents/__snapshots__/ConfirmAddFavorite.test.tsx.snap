// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <ConfirmBlock /> 1`] = `
<div
  className="sc-gqjmRU hSHDLZ"
>
  <div
    className="sc-gzVnrw gGucts"
  >
    <div
      className="sc-EHOje emHEG"
    >
      Add this user?
    </div>
  </div>
  <div
    className="sc-jTzLTM zTzxT"
  >
    This user's information will be added to your favorite users' list.
  </div>
  <div
    className="sc-VigVT jaAGru"
  >
    <button
      className="sc-bdVaJa bsJYMs"
      data-active={true}
      fontSize={40}
      height={112}
      onClick={[Function]}
      type="red"
      width={272}
    >
      Cancel
    </button>
    <button
      className="sc-bdVaJa cLgmoo"
      data-active={true}
      fontSize={40}
      height={112}
      onClick={[Function]}
      type="white"
      width={272}
    >
      OK
    </button>
  </div>
</div>
`;
